extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -161,6 +163,7 @@
   private final Condition shouldResume = pauseLock.newCondition();
   private final Lock pollRetryLock = new ReentrantLock();
   private final Condition isAwaitingRetry = pollRetryLock.newCondition();
+  private final Object statusLock = new Object();
```

Comment:
@gianm there's a big comment just above these lines that describes what most of the locks and conditions do, I'll add more details for pollRetryLock and statusLock